controllers/azurecluster_controller.go

```diff
@@ -296,3 +297,46 @@ func (acr *AzureClusterReconciler) reconcileDelete(ctx context.Context, clusterS
 
 	return reconcile.Result{}, nil
 }
+
+func clusterToInfrastructureMapFunc(ctx context.Context, c client.Client, controllerName string) handler.MapFunc {
+	return func(o client.Object) []ctrl.Request {
+		ctx, log, done := tele.StartSpanWithLogger(ctx, "controllers."+controllerName+".clusterToInfrastructureMapFunc")
```

Comment:
this function is in the azurecluster_controller file, if it's meant to be used by multiple controllers should it be somewhere common instead? maybe helpers.go?

Also side question, if this needed for other providers too (sounds like it was added to CAPA), why not change the `ClusterToInfrastructureMapFunc` function in CAPI directly to skip externally managed clusters instead of overriding it with our own `clusterToInfrastructureMapFunc` func?

Reply:
Good point, opened https://github.com/kubernetes-sigs/cluster-api/pull/6039 for that and will update this and capa once it merges

Reply:
Updated to bump cluster-api and use the new ClusterToInfrastructureMapFuncWithExternallyManagedCheck, please have another look.